### PR TITLE
Fixing memory leak in mdfind table

### PR DESCRIPTION
**Pull request by mkareta**
CFObjects were created but never released.

### CODE REVIEW

**Comment by osqueryer** [comment]
:no_good: The commit 2d800259596fa95e2fd6418bc3dd08c726768f81 (Job results: [3629](https://jenkins.osquery.io/job/osqueryPullRequestAudit/3629/)) failed the code audit or documentation test.

**Comment by obelisk** [comment]
I ran leaks on this but found none, how did you find and verify these leaks? :)

**Comment by fmanco** [comment]
Somehow the tests for osx are on triggering. Lets try again: ok to test.

File: osquery/tables/system/darwin/mdfind.mm [hunk 1]
```diff
@@ -107,12 +116,17 @@ Status waitForSpotlight(const std::vector<NamedQuery>& queries) {
...
+  
   if (!waitForSpotlight(queries).ok()) {
+    releaseQueries(queries);
     return results;
   }
 
+  genResults(queries, results);
+  releaseQueries(queries);
+
   genResults(queries, results);
```
Comment (fmanco) [inline osquery/tables/system/darwin/mdfind.mm]: Did you forget to delete this?

**Comment by osqueryer** [comment]
:no_good: The commit 2d800259596fa95e2fd6418bc3dd08c726768f81 (Job results: [3631](https://jenkins.osquery.io/job/osqueryPullRequestAudit/3631/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:-1: The commit 2d800259596fa95e2fd6418bc3dd08c726768f81 (Job results: [7180](https://jenkins.osquery.io/job/osqueryPullRequestLinux/7180/)) failed one or more tests (**Linux**).

**Comment by osqueryer** [comment]
:no_good: The commit d73e293e10e64e6d784e53f035d54599f6d2da02 (Job results: [3636](https://jenkins.osquery.io/job/osqueryPullRequestAudit/3636/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:-1: The commit aba052628ebaf9bd57bb4b4b8fd3b4bbcb0288cd (Job results: [7194](https://jenkins.osquery.io/job/osqueryPullRequestLinux/7194/)) failed one or more tests (**Linux**).

**Pull request merged**